### DISCUSSION TITLE
Azure Custom Installation

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#684] [General] [answered]
thanks so much for making the "deploy to Azure" button.  I'm setting that up as our sandbox to get other working on the CMS side of things.  I'm courious about the Blazor SKU and the database fields.
- will this create a Sql Server and database?
- the blazor tiers looks the same as app service tiers?

![image](https://user-images.githubusercontent.com/311063/90301037-471d2b80-de52-11ea-9955-850e70eb9255.png)

**Comment by markdav-is** [comment]
I've tried 3 times to use the Deply to Azure button today and it always fails to create the wed site.   Maybe @ADefWebserver or @mikecasas have some tips.  I'm going to try re-deploying the web sit on top of the web that was created. 

[Deployment-Microsoft.Template.zip](https://github.com/oqtane/oqtane.framework/files/5081761/Deployment-Microsoft.Template.zip)

**Reply by mikecasas** [reply]
I haven't tried the Azure install button. I followed @ADefWebserver blog post at https://blazorhelpwebsite.com/ViewBlogPost/1 to get online with Azure.

**Comment by ADefWebserver** [comment, marked as answer]
@nohorse Please see if the directions help: [Oqtane Deploy to Azure.docx](https://github.com/oqtane/oqtane.framework/files/4956136/Oqtane.Deploy.to.Azure.docx) or: https://blazorhelpwebsite.com/ViewBlogPost/40

Pay close attention to the sample values. I have found that the reason a deployment failed was giving each resource a name that wasn't acceptable by Azure or not having a password that was strong enough.

Unfortunately, the files you attached indicate there is an error, but, not the exact error. To get the exact error, you can click on the Deployment error in the Azure website (see: [Troubleshoot common Azure deployment errors with Azure Resource Manager](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/common-deployment-errors)).

The Deploy to Azure button creates the SQL server and the website needed, basically everything.

**Reply by markdav-is** [reply]
those are indeed the docs I'm following and it pauses at deploying the web and that's the part that fails.  I'll try it again from scratch just to be very shure.
![image](https://user-images.githubusercontent.com/311063/90416365-3a255580-e067-11ea-9ac2-66ee31c92a32.png)

**Comment by ADefWebserver** [comment]
@nohorse In your screenshot I am not seeing any failures?
It does take a long time, perhaps you just need to wait?

**Reply by markdav-is** [reply]
I was able to get passed things, buy deploying from VS to azure on top of the one that was created.  I never found out exactly why it failed.  thanks for your help getting unblocked.
